### ISSUE TITLE
pg_test flakes

### ISSUE THREAD

**Issue by sploiselle** [MaterializeInc/materialize#14384]
### What version of Materialize are you using?

main

### How did you install Materialize?

Built from source

### What is the issue?

`pg_test` flakes on me regularly. Here's an example where I had to run 4 times to get a pass:
https://buildkite.com/materialize/tests/builds/40048#0182d0c4-6dae-4dfe-b6fe-bfac159815fb

I also cannot get `pg_test` to pass locally on my machine.

### Relevant log output

```source
'test_pgtest' panicked at 'called `Result::unwrap()` on an `Err` value: Resource temporarily unavailable (os error 11): waiting for ReadyForQuery, saw [\n    \"CommandComplete {\\\"tag\\\":\\\"DROP TABLE\\\"}\",\n    \"ReadyForQuery {\\\"status\\\":\\\"I\\\"}\",\n    \"CommandComplete {\\\"tag\\\":\\\"CREATE TABLE\\\"}\",\n    \"ReadyForQuery {\\\"status\\\":\\\"I\\\"}\",\n    \"ParameterDescription {\\\"parameters\\\":[]}\",\n    \"RowDescription {\\\"fields\\\":[{\\\"name\\\":\\\"a\\\"}]}\",\n    \"ReadyForQuery {\\\"status\\\":\\\"I\\\"}\",\n]
```
```

**Comment by mjibson** [comment]
I'm unable to reproduce this locally. I tried: `while true; do cargo test --package mz-environmentd --test pgwire -- test_pgtest --exact --nocapture; done` and ran it for a few hours. One idea: we need to add even more time than the 30s timeout? That seems odd?

**Comment by benesch** [comment]
Seems to have been fixed by #14462.
